src/Microsoft.AspNetCore.Routing/DefaultLinkGenerator.cs

```diff
@@ -92,7 +92,9 @@ private string GetLink(MatcherEndpoint endpoint, LinkGeneratorContext context)
 
             var templateValuesResult = templateBinder.GetValues(
                 ambientValues: context.AmbientValues,
-                values: context.ExplicitValues);
+                explicitValues: context.ExplicitValues,
+                endpoint.RequiredValues.Keys);
```

Comment:
How does this compile? It's a positional argument following named args. Did that make that allowed?

Reply:
Oh DOPE: https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-2#non-trailing-named-arguments

they made that allowed